### ISSUE TITLE
EvidenceLine (taking another shot at this concept)

### ISSUE THREAD

**Issue by larrybabb** [clingen-data-model/clingen-interpretation#39]
When we assemble several structured data elements into a complex statement of evidence for the explicit purpose of producing a summary of information which can then be used to make a basis for an assertion. 

It is the curators/assessors/interpreters way of summarizing in a single statement the information on which their assertion is made.

Right now we build into the CriterionAssessment and VariantInterpretation an "explanation" or "description" attribute which truly represents this concept.  However, doesn't this "evidenceLine" or "evidenceStatement" concept need to exist before the actual assertion gets made and therefore should conceptually be pulled out of the "outcome" of the assertion process whether it be a CaptureEvidence, AssessCriterion or GenerateInterpretation?

I believe there will be cases when an EvidenceLine will be needed to capture nuanced information and the assembled (underlying) data which in itself represents the culmination of that which is being asserted. Thus, the EvidenceLine (or evidence statement) is the root concept that is input to any assertion and it is critical at providing access to the various types of associated data that provides the information to produce such an evidenceLine.

In the context of a CriterionAssessment it is somewhat easy to see this "joining" of data to produce an overarching line or statement of evidence.  In regards to VariantInterpretation, I think the EvidenceLine is the fully assembled statement that will then be used to publish a narrative paragraph (or two) and it would include a comprehensive but summary representation of all of the indivudally assessed criterion and their respective evidenceLines w/ structuredData.

**Comment by bpow** [comment]
Thanks for starting this discussion, it's one we've needed for a while because I don't think we have a shared mental model of what EvidenceLine actually represents. Hopefully @mbrush can comment as well...

I agree that it is important to try to capture nuanced information in the way that you are describing for multiple reasons-- it provides a "break the glass" in case the analyst finds that there is something s/he wants to express that is not easily represented in the model, but secondarily, having a free text field like this (as much as free text is the bane of structured data) allows us to bring forward legacy data.

However, that's not necessarily how I've been thinking of the "explanation" or "description" fields. I've seen these values, as we have them now, as a form of documentation whereby within our examples we provide the reader with a free-form description of what the structured form is supposed to represent. From that standpoint, these fields would clearly not be required in implementation (if the model is truly adequate to represent what it is supposed to).

What you are describing in your last paragraph sounds almost like we would be re-generating a textual form of the interpretation through the use of templates. I think this would be a useful tool, but I don't think that it is in scope for the data model itself.

## So what do I think `EvidenceLine` means?

I think that my idea of what a "line of evidence" is differs from your thinking fairly significantly. This is something we need to work out in order to figure out how we should map to the SEPIO `EvidenceLine`.

I hadn't really thought much about the expression "line of evidence" prior to our interactions with SEPIO, so to try to get some precision to how other's might interpret it, I did a web search... The best contextualization of this phrase that I could come up with is as part of a [consilience](https://en.wikipedia.org/wiki/Consilience). In this manner of thinking, each "line of evidence" is an assessment of a hypothesis through somewhat-independent forms of evidence. The wikipedia consilience entry speaks of differents methods of measurements (using a meter stick, laser TOF, triangulation to stars), not just different measurements.

The SEPIO documentation states "*Evidence Lines* represent evidence for a proposition that is derived from a single line of inquiry."

Mapping this to variant interpretation, I would think that things like rare allele frequency, presence of an allele in _de novo_ state in an individual with no family history of the condition, predicted protein truncation resulting from the variant, etc. all represent independent lines of evidence (arising from independent lines of inquiry).

In my mind, these lines of evidence map fairly well to the ACMG variant interpretation criteria (probably because without thinking of it in these terms, the authors were trying to define the independent lines of inquiry that they use to interpret variants.

**Comment by larrybabb** [comment]
> In my mind, these **lines of evidence map fairly well to the ACMG variant interpretation criteria** (probably because without thinking of it in these terms, the authors were trying to define the independent lines of inquiry that they use to interpret variants.

I think the lines of evidence (input or basis) of a variant interpretation can be made up not only of individual assessed criteria (the data and the assessment, not just the criteria) but also some additional factors which ultimately lead an interpreter (geneticists/physician/etc) to arrive at an final assertion or published interpretation.  I think it is idealistic to assume the only factors that can contribute to an interpretation are the assessed criterion, even thought that is the lions share for sure. 

A geneticist or interpreter carefully crafts the argument as a narrative, primarily based on assessed criterion, which is essentially the closing argument on which the final assessment is made.  I think it is reasonable to consider these arguments or explanations to be the wrapper classes which coincide with what SEPIO is calling evidenceLine.

Very interesting discussion.  We do need to support a large scope of situations and direct less structured approaches to more structured ones in order to provide more computational support.  But I don't think it is our job to be overly zealous about the structure so that it raises the bar too high on adoption.  

I am attempting to build a system in parallel. And, we see the need for an evidence statement or line which provides a "grouping" of sorts for the structured data prior to it being assessed on a criterion-by-criterion basis.  I do have more difficulty seeing the benefit of the evidence line at the interpretation level, but I can sort of see the pattern fit when I think of the summarization of all the assessed criterion (plus any other less definable points) being written out by the interpreter to provide a complete and comprehensive statement which provides the necessary information for understanding the interpreters final assertion.

Good discussion.

**Comment by bpow** [comment]
So is what you want to capture additional factors that are outside of defined criteria, or additional factors that modify the assessment of the criteria (e.g., this would be PM2, but I'm going to count it as PP because...)? In the former case, we would need a separate item (`EvidenceLine` or something else?) that allows that data to be captured. In the latter, the additional modifications belong within the `CriterionAssessment` itself

If `EvidenceLine` is to represent the "closing statement", then I think it is confusingly named. I don't think that a closing statement is a line of evidence, but that a closing statement would draw upon multiple lines of evidence ("We showed that the defendant was in dire financial straits, establishing motive. We showed that he was in the area of the bank at 3am, consistent with when the vault was entered. We showed that he deposited a large sum of money at a different bank the next day, etc... Putting these lines of evidence together, it is clear that the defendant is guilty...")

**Comment by larrybabb** [comment]
I am not necessarily trying to force this concept nor do I necessarily have a good handle on what SEPIO means by it.  

I am finding in the implementation we are building that there is a need for some concept that would group data items together to make a cohesive statement before a criterion assessment can be made.  I do submit that the outcome of the assessed criterion activity in combination with its assertion method and the proposition (Var1 Path for DCM) provides the input for the validated assertion which we know as the interpretation.   So, for the interpretation type assertion I would concede that each of the assessed criterion activity outputs would fit the concept of an evidence line for input to the final interpretation activity (which obviously is a bit different than what I perceive Matt B has graphically represented thus far).

**Comment by bpow** [comment]
My thought has been that the `Criteria` themselves are the grouping of the data and the description of how the grouping takes place (e.g. this variant has allele frequency 0.0003 and this condition is expected to result from alleles with frequency no greater than 0.001)... Our model doesn't directly specify how the data are combined other than in the description of each `Criterion`.  Maybe we could be more explicit about the "standard" implementation of the criteria, but I'm not sure how we would without making some sort of rule language.

Where people want to make an assertion that differs from the commonly-accepted use of a `Criterion`, they should have a way to do so (by a free-text field, tied to the person who is making the assertion). Is that closer to what you are trying to get at?

**Comment by larrybabb** [comment]
That sounds conceptually right. I'd like to explore it with some modeling adjustments to be sure.

**Comment by cbizon** [comment]
Sorry, I just got around to checking out this very interesting discussion.  Larry - maybe you could provide some examples where you see this grouping as helpful.   I'm having a little trouble grasping what you're going for here.

**Comment by bpow** [comment]
We had some discussion about this on a conference call this morning. The quick summary (and @larrybabb can clarify/expand as needed) is that there are two related items.

1. We are still not clear on how to best utilize the `EvidenceLine` concept from SEPIO, but will discuss with @mbrush tomorrow on that conference call. It would especially be useful to have a concrete representation of some real-world data to help form our mental models of this concept.
2. @larrybabb wants to make sure that we have some flexibility/escape valve to capture evidence that is not well-structured (e.g., to bring in legacy data) or structured differently (e.g., new evidence types not yet incorporated into the model. we discussed a few options for how to do this:
   1. we could use an existing free-text field for this purpose
   2. we could add an additional free-text field to some high level object (e.g. Data) that would be expressly for this, or
   3. we could make an "UnstructuredData" object that would have some free-text field and would be permitted to be associated with any CriterionAssessment.

There was some preference for option (iii) on the call, but with the idea that we would do some modeling/diagramming to see how this would work out.

**Comment by larrybabb** [comment]
Started a new issue which narrows focus and considers some more recent developments see #40

**Comment by larrybabb** [comment]
Closing based on our fairly successful work in harmonizing with SEPIO.  There may still be residual work here. But I will close for now and others may re-open in the event that it is needed.
